scene/gui/file_dialog.cpp

```diff
@@ -1103,8 +1103,15 @@ void FileDialog::set_access(Access p_access) {
 	if (access == p_access) {
 		return;
 	}
+	access = p_access;
+	root_prefix = "";
+	root_subfolder = "";
+
 	switch (p_access) {
 		case ACCESS_FILESYSTEM: {
+#ifdef ANDROID_ENABLED
+			set_root_subfolder(OS::get_singleton()->get_system_dir(OS::SYSTEM_DIR_DESKTOP));
```

Comment:
`set_root_subfolder(...)` makes use of `dir_access`, so let's invoke it after setting `dir_access`.